utils/error_injection.hh

```diff
@@ -371,22 +371,22 @@ public:
     }
 
     // \brief Inject exception
-    // \param exception_factory function returning an exception pointer
-    template <typename Func>
-    requires std::is_invocable_r_v<std::exception_ptr, Func>
+    template <typename Exception>
     [[gnu::always_inline]]
-    future<>
-    inject(const std::string_view& name,
-            Func&& exception_factory) {
+    void inject(const std::string_view& name) {
```

Comment:
Let's constrain Exception by ensuring it inherits from std::exception and can be constructible from const char* or nothing.

Reply:
Good point, fixed